File: scala/134.scala
object Solution {
    def canCompleteCircuit(gas: Array[Int], cost: Array[Int]): Int = {
        var l = 0
        var r = gas.length - 1
        var g = 0
        while (l < r) {
            if (g + gas(l) < cost(l)) {
                g += gas(r)-cost(r)
                r -= 1
            } else {
                g += gas(l) - cost(l)
                l += 1
            }
        }
        if (g + gas(l) < cost(l)) -1 
        else (l+1) % gas.length
    }
}

File: leetcodeJava/134.java
package leetcodeJava;

class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        for (var i = 0; i < gas.length; i++) {
            var cur = 0;
            for (var j = i; j < i+gas.length + 1; j++) {
                if (j == i+gas.length) return i;
                var _j = j % gas.length;
                cur += gas[_j];
                if (cur < cost[_j]) {
                    if (_j < i) return -1;
                    else i = _j;
                    break;
                }
                cur -= cost[_j];
            }
        }
        return -1;
    }
}
